.github/workflows/pr-wasm-preview.yml

```diff
@@ -0,0 +1,221 @@
+name: PR WASM Preview Deployment
+
+on:
+  pull_request:
+    branches:
+      - main
+      - dev
+    types:
+      - opened
+      - synchronize
+      - reopened
+      - labeled
+      - unlabeled
+
+env:
+  JEMALLOC_SYS_WITH_MALLOC_CONF: "background_thread:true,narenas:1,tcache:false,dirty_decay_ms:0,muzzy_decay_ms:0,metadata_thp:auto"
+
+jobs:
+  build-deploy:
+    timeout-minutes: 60
+    runs-on: ubuntu-latest
+    container: komodoofficial/ci-container:latest
+    steps:
+      - name: Check label
+        id: check
+        run: |
+          has_label=$(jq -r '.pull_request.labels[].name' "$GITHUB_EVENT_PATH" | grep -x 'deploy: wasm-playground' || true)
+          if [ -n "$has_label" ]; then
+            echo "should_run=true" >> $GITHUB_OUTPUT
+          else
+            echo "should_run=false" >> $GITHUB_OUTPUT
+          fi
```

Comment:
Prefer a job-level `if` that checks for the label, this way the job won’t even start unless the label is present which is cleaner than starting and exiting early.

Reply:
done